The `max_model_length` of bge-small-en being 1000000000000000019884624838656 overflows in the Actions environment. "Works on my machine" moment here.
```
Model is local_cache/fast-bge-small-en with max length -9223372036854775808
--- FAIL: TestEmbedBGESmallEN (2.36s)
    fastembed_test.go:66: Expected no error, got The tensor's shape ([1 512]) requires 512 elements, but only 8 were provided
```

--- a/fastembed.go
+++ b/fastembed.go
@@ -334,9 +334,14 @@ func loadTokenizer(modelPath string, maxLength int) (*tokenizer.Tokenizer, error
 		return nil, err
 	}
 
-	maxLength = min(maxLength, int(tokenizerConfig["model_max_length"].(float64)))
+	modelMaxLen := tokenizerConfig["model_max_length"].(float64)
+	// Handle overflow on Linux
+	if modelMaxLen > math.MaxInt {
+		modelMaxLen = math.MaxInt
+	}
 
-	fmt.Println("Model is", modelPath, "with max length", maxLength)
+	maxLength = min(maxLength, int(modelMaxLen))
+
 	tknzer.WithTruncation(&tokenizer.TruncationParams{
 		MaxLength: maxLength,
 		Strategy:  tokenizer.LongestFirst,

--- a/fastembed.go
+++ b/fastembed.go
@@ -335,14 +335,7 @@ func loadTokenizer(modelPath string, maxLength int) (*tokenizer.Tokenizer, error
 	}
 
 	modelMaxLen := tokenizerConfig["model_max_length"].(float64)
-	// Handle overflow on Linux
-	if modelMaxLen > math.MaxInt {
-		modelMaxLen = math.MaxInt
-	}
-
-	maxLength = min(maxLength, int(modelMaxLen))
-
-	fmt.Println("Model is", modelPath, "with max length", maxLength)
+	maxLength = min(maxLength, int(math.Abs(modelMaxLen)))
 
 	tknzer.WithTruncation(&tokenizer.TruncationParams{
 		MaxLength: maxLength,

--- a/fastembed.go
+++ b/fastembed.go
@@ -342,6 +342,8 @@ func loadTokenizer(modelPath string, maxLength int) (*tokenizer.Tokenizer, error
 
 	maxLength = min(maxLength, int(modelMaxLen))
 
+	fmt.Println("Model is", modelPath, "with max length", maxLength)
+
 	tknzer.WithTruncation(&tokenizer.TruncationParams{
 		MaxLength: maxLength,
 		Strategy:  tokenizer.LongestFirst,

